pkg/apis/apps/validation/validation.go

```diff
@@ -63,10 +66,10 @@ func ValidatePodTemplateSpecForStatefulSet(template *api.PodTemplateSpec, select
 				allErrs = append(allErrs, field.Invalid(fldPath.Child("metadata", "labels"), template.Labels, "`selector` does not match template `labels`"))
 			}
 		}
-		// TODO: Add validation for PodSpec, currently this will check volumes, which we know will
-		// fail. We should really check that the union of the given volumes and volumeClaims match
-		// volume mounts in the containers.
-		// allErrs = append(allErrs, apivalidation.ValidatePodTemplateSpec(template, fldPath)...)
+		podSpecErrs := apivalidation.ValidatePodTemplateSpec(template, fldPath, opts)
+		if len(podSpecErrs) > 0 && !setOpts.SkipValidatePodTemplateSpec {
+			allErrs = append(allErrs, podSpecErrs...)
```

Comment:
check the boolean first to avoid doing work we'll throw away

```suggestion
		if !setOpts.SkipValidatePodTemplateSpec {
			allErrs = append(allErrs, apivalidation.ValidatePodTemplateSpec(template, fldPath, opts)...)
```